### PR TITLE
Remove assumptions about preprocessor_next

### PR DESCRIPTION
**Pull request by AnirudhNarsipur**
Slight changes to the prototype that remove making any assumptions about the preprocessor (previously it was assumed there were no false negatives).

### CODE REVIEW

File: model-checking/scheduler.als [hunk 1]
```diff
@@ -74,6 +105,8 @@ sig Command {
         after (c.command_state = NE)
 
         (some nonCommittedDependencies[c, (~preprocessor_next)])
+
```
Comment (sidprasad) [inline model-checking/scheduler.als]: nit: Remove

File: model-checking/scheduler.als [hunk 2]
```diff
@@ -90,12 +123,16 @@ sig Command {
         after (c.command_state = NE)
 
         (some nonCommittedDependencies[c, dependency])
+
+        //TODO: shouldn't we have to specify something here?
```
Comment (sidprasad) [inline model-checking/scheduler.als]: I think this is the key situation (and is tricky)

If there is a dependency, you need to add it to preprocessor_next, but also remove any dependencies from preprocessor_next that would violate partialOrder

File: model-checking/scheduler.als [hunk 3]
```diff
@@ -58,12 +75,26 @@ sig Command {
...
         {(c.command_state) = C} implies always (c.command_state = C)
         {(c.command_state) = CN} implies always (c.command_state = CN)
+
     }
 
     fun nonCommittedDependencies[c : Command, previous : set (Command -> Command)] : set Command
     {
         {x : Command | (x in c.^previous) and !committed[x] }
     }
+    pred nonCommittedDep[c : Command]
```
Comment (sidprasad) [inline model-checking/scheduler.als]: What should this predicate be used for? I don't see it currently referenced

File: model-checking/scheduler.als [hunk 4]
```diff
@@ -35,8 +35,25 @@ sig Command {
...
+                let backdep = (c.preprocessor_next' - c.preprocessor_next)  | {
+                    trace_executor_found_dependency[backdep]
+                }
+                
+                let forwarddep = (preprocessor_next'.c - preprocessor_next.c)  | {
+                    trace_executor_found_dependency[forwarddep]
+                }
+            }
+       }
+
```
Comment (sidprasad) [inline model-checking/scheduler.als]: Can you help me understand the intent here :) Comments may be useful

Comment (sidprasad) [reply model-checking/scheduler.als]: I may be misunderstanding, but this doesn't seem like a property of wellFormed-ness

**Review by sidprasad — CHANGES_REQUESTED** [review]

File: model-checking/scheduler.als [hunk 5]
```diff
@@ -24,21 +24,74 @@ sig Command {
 
     pred partialOrder[r: set (Command -> Command)] {
         no r & iden // Anti-reflexivity
-        all x: Command, y: Command, z: Command |
+        all x: Command, y: Command |
         {
-            (x->y in r and y->z in r) implies x->z in r // Transitivity
+            (x->y in r) implies no (((y.^r)->x) & r) // Transitivity
```
Comment (sidprasad) [inline model-checking/scheduler.als]: This is not transitivity. What's the intended change here?

This predicate no longer represents a partial order (it's over constrained)

**Comment by AnirudhNarsipur** [comment]
Newest Commits should have a working model of variable preproc. 

See inline comments.

File: model-checking/scheduler.als [hunk 6]
```diff
@@ -167,3 +200,21 @@ check {final implies (always final)  } for exactly 4 State , 6 Command
...
+run {
+    scheduler_e2e 
+    some dependency
+    some c1 ,c2 : Command {
+         (c1->c2) in preprocessor_next
+         (c1->c2) in ~dependency
+         
+        eventually( not (c1->c2 in preprocessor_next))
+    }
+}  for exactly 4 State , 6 Command
```
Comment (sidprasad) [inline model-checking/scheduler.als:220]: nit: phrase this to be a valid check or run

**Review by sidprasad — APPROVED** [review]

**Pull request merged**